tools/monthly-updates-blog-post/generate-blog-post.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+
+import calendar
+import json
+import os
+import re
+import sys
+from datetime import date, datetime
+from textwrap import dedent
+from urllib.parse import urlencode
+
+import requests
+
+today = date.today()
+year = today.year
+month = today.month - 1 if today.day < 10 else today.month
+
+if month == 0:
+    month = 12
+    year = year - 1
+
+month_name = calendar.month_name[month]
+
+
+def strip_html_tags(text: str):
+    text = re.sub(r"<[^<]+?>", "", text)
+    return re.sub(r"\s+|\\n|&nbsp;", " ", text).strip()
+
+
+def get_github_pulse(repo: str):
+    # Get the GitHub Pulse summary for the repo
+    response = requests.get(
+        f"https://github.com/zaproxy/{repo}/pulse_diffstat_summary?period=monthly"
+    )
+    return strip_html_tags(response.text)
+
+
+def get_human_pr_count_summary():
+    query = {
+        "q": f"org:zaproxy type:pr -author:zapbot -author:app/dependabot sort:updated-asc closed:{year}-{month:0>2} is:merged",
+        "type": "pullrequests",
+    }
+    query_string = urlencode(query)
+    response = requests.get(f"https://github.com/search/count?{query_string}")
+    count = strip_html_tags(response.text)
+    return f"A total of [{count} human PRs were merged](https://github.com/search?{query_string}) on the ZAP repos."
+
+
+def get_release_notes():
+    # Get zap-extensions releases using GitHub API
+    headers = {"X-GitHub-Api-Version": "2022-11-28"}
+    if os.environ.get("GITHUB_TOKEN"):
+        headers["Authorization"] = f"Bearer {os.environ['GITHUB_TOKEN']}"
+    response = requests.get(
+        "https://api.github.com/repos/zaproxy/zap-extensions/releases?per_page=200"
```

Comment:
Just to not be misleading I'd suggest setting back to 100 as that's the max allowed: https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#list-releases--parameters